wordle/src/main/kotlin/domain/Word.kt

```diff
@@ -0,0 +1,86 @@
+package domain
+
+class Word(word: String) {
+
+    private val alphabets: List<Alphabet> = word.map { Alphabet(it) }
+
+    init {
+        require(alphabets.size == Wordle.WORD_LENGTH) { "단어는 ${Wordle.WORD_LENGTH} 글자로 구성됩니다" }
+    }
+
+    fun compare(other: Word): List<Compared> {
+        val result = MutableList(Wordle.WORD_LENGTH) { Compared.NONE }
+        val pool = AlphabetsPool(alphabets)
+
+        for ((i, alphabet) in other.alphabets.withIndex()) {
+            if (alphabets[i] == alphabet) {
+                result[i] = Compared.EQUAL
+                pool.spend(alphabet)
+                continue
+            }
+
+            if (alphabets.contains(alphabet) && pool.has(alphabet)) {
+                result[i] = Compared.EXIST
+                pool.spend(alphabet)
+            }
+        }
+        return result
+    }
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as Word
+
+        return (alphabets == other.alphabets)
+    }
+
+    override fun hashCode(): Int {
+        return alphabets.hashCode()
+    }
+
+    override fun toString(): String {
+        return alphabets.map { it.value }.joinToString(separator = "")
```

Comment:
```suggestion
        return word
```
👀

만약 word가 프로퍼티라면 훨씬 간결해지고 빨라지겠군요